src/peft/tuners/shira/config.py

```diff
@@ -0,0 +1,128 @@
+    )
+    mask_type: Literal["random"] = field(
+        default="random",
+        metadata={
+            "help": (
+                "Type of mask function. Defaults to a random sparse mask. "
+                "An optional user-defined mask_fn to compute the mask value can also be supplied by instantiating `config = ShiraConfig(...)` and then setting "
+                "`config.mask_fn = <your custom mask function>`. For a pretrained weight with shape m x n, the custom mask function must return only one mask (shape: m x n) "
+                "which must be binary 0 or 1 with num_shira_parameters = r(m + n) for linear layers. Device and dtype of mask must be same as base layer's weight's device and dtype. "
+                "Please see mask_functions.py for more details and to see the default random sparse mask implementation."
+            )
+        },
+    )
+    random_seed: Optional[int] = field(
+        default=None, metadata={"help": "random seed for the torch generator for random_mask"}
+    )
+    target_modules: Optional[Union[list[str], str]] = field(
+        default=None,
+        metadata={
+            "help": (
+                "List of module names or regex expression of the module names to replace with SHiRA."
+                "For example, ['q', 'v'] or '.*decoder.*(SelfAttention|EncDecAttention).*(q|v)$'. "
+                "Only linear layers are supported."
+            )
+        },
+    )
+    fan_in_fan_out: bool = field(
+        default=False,
+        metadata={"help": "Set this to True if the layer to replace stores weight like (fan_in, fan_out)"},
+    )
+    init_weights: bool = field(
+        default=True,
+        metadata={
+            "help": "Initialize SHiRA weight to have zero values. If set to False, SHiRA weights are initialized to randn values instead of zeros and this is used only for testing."
+        },
+    )
+    modules_to_save: Optional[list[str]] = field(
+        default=None,
+        metadata={
+            "help": (
+                "List of modules apart from SHiRA layers to be set as trainable and saved in the final checkpoint. For"
+                " example, in Sequence Classification or Token Classification tasks, the final layer"
+                " `classifier/score` are randomly initialized and as such need to be trainable and saved."
+            )
+        },
+    )
+
+    def __post_init__(self):
+        super().__post_init__()
+        self.peft_type = PeftType.SHIRA
+        self.target_modules = (
+            set(self.target_modules) if isinstance(self.target_modules, list) else self.target_modules
+        )
+        if self.mask_type == "random":
+            self.mask_fn = random_mask
+        else:
+            warnings.warn(
+                f"Argument {self.mask_type=} is not recognized, please supply your own masking function by calling `config.mask_fn = my_mask_fn`. Please ignore this warning message if you are loading from a pretrained checkpoint."
+            )
```

Comment:
I think this should be a good way to avoid showing the warning if not needed:

```suggestion
            if not self.inference_mode:
                warnings.warn(
                    f"Argument {self.mask_type=} is not recognized, please supply your own masking function by calling "
                    "`config.mask_fn = my_mask_fn`. Please ignore this warning message if you are loading from a "
                    "pretrained checkpoint."
                )
```

PEFT automatically sets `inference_mode` on the config when loading a PEFT model:

https://github.com/huggingface/peft/blob/35000fda88fdf29db785783f8a40431cee2cab10/src/peft/peft_model.py#L450

Reply:
Very good suggestion. We have added this now and it works. Thanks.